internal/appsec/waf_test.go

```diff
@@ -14,13 +14,35 @@ import (
 	"testing"
 
 	pAppsec "github.com/DataDog/dd-trace-go/v2/appsec"
-	httptrace "github.com/DataDog/dd-trace-go/v2/contrib/net/http"
 	"github.com/DataDog/dd-trace-go/v2/ddtrace/mocktracer"
 	"github.com/DataDog/dd-trace-go/v2/internal/appsec"
-
+	"github.com/DataDog/dd-trace-go/v2/internal/appsec/dyngo/instrumentation/httpsec"
+	"github.com/DataDog/dd-trace-go/v2/internal/contrib/httptrace"
 	"github.com/stretchr/testify/require"
 )
 
+type mockServeMux struct {
+	*http.ServeMux
+}
+
+func (mux *mockServeMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	span, ctx := httptrace.StartRequestSpan(r)
+	defer func() {
+		httptrace.FinishRequestSpan(span, http.StatusOK)
+	}()
+	h := http.Handler(mux.ServeMux)
+	if appsec.Enabled() {
+		h = httpsec.WrapHandler(h, span, make(map[string]string))
+	}
+	h.ServeHTTP(w, r.WithContext(ctx))
+}
+
+func newMockServeMux() *mockServeMux {
+	return &mockServeMux{
+		http.NewServeMux(),
+	}
+}
+
```

Comment:
This was implemented to break a circular dependency between dd-trace-go and a contrib.